third_party/move/move-compiler-v2/tests/checking/typing/while_condition_invalid.exp

```diff
@@ -0,0 +1,49 @@
+
+Diagnostics:
+error: expected `bool` but found `()` in expression
```

Comment:
I still think it would be better to say "expected `bool` but found expression of type `()`"

Reply:
Its not always an expression. It can be a pattern or a constant as well. It depends on the context.